src/utils/viewer/simviewer.cpp

```diff
@@ -0,0 +1,669 @@
+#else
+  args.addArgument("scene")
+#endif
+      .setHelp("scene", "scene file to load")
+      .addSkippedPrefix("magnum", "engine-specific options")
+      .setGlobalHelp("Displays a 3D scene file provided on command line")
+      .addBooleanOption("enable-physics")
+      .addBooleanOption("scene-requires-lighting")
+      .setHelp("scene-requires-lighting", "scene requires lighting")
+      .addBooleanOption("debug-bullet")
+      .setHelp("debug-bullet", "render Bullet physics debug wireframes")
+      .addOption("physics-config", ESP_DEFAULT_PHYS_SCENE_CONFIG_REL_PATH)
+      .setHelp("physics-config", "physics scene config file")
+      .addOption("navmesh-file")
+      .setHelp("navmesh-file", "manual override path to scene navmesh file")
+      .addBooleanOption("recompute-navmesh")
+      .setHelp("recompute-navmesh", "programmatically generate scene navmesh")
+      .parse(arguments.argc, arguments.argv);
+
+  const auto viewportSize = Mn::GL::defaultFramebuffer.viewport().size();
+
+  imgui_ =
+      Mn::ImGuiIntegration::Context(Mn::Vector2{windowSize()} / dpiScaling(),
+                                    windowSize(), framebufferSize());
+
+  /* Set up proper blending to be used by ImGui. There's a great chance
+     you'll need this exact behavior for the rest of your scene. If not, set
+     this only for the drawFrame() call. */
+  Mn::GL::Renderer::setBlendEquation(Mn::GL::Renderer::BlendEquation::Add,
+                                     Mn::GL::Renderer::BlendEquation::Add);
+  Mn::GL::Renderer::setBlendFunction(
+      Mn::GL::Renderer::BlendFunction::SourceAlpha,
+      Mn::GL::Renderer::BlendFunction::OneMinusSourceAlpha);
+
+  // Setup renderer and shader defaults
+  Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::DepthTest);
+  Mn::GL::Renderer::enable(Mn::GL::Renderer::Feature::FaceCulling);
+
+  sceneFileName = args.value("scene");
+  bool useBullet = args.isSet("enable-physics");
+  if (useBullet && (args.isSet("debug-bullet"))) {
+    debugBullet_ = true;
+  }
+
+  // configure and intialize Simulator
+  auto simConfig = esp::sim::SimulatorConfiguration();
+  simConfig.scene.id = sceneFileName;
+  simConfig.enablePhysics = useBullet;
+  simConfig.frustumCulling = true;
+
+  simulator_ = esp::sim::Simulator::create_unique(simConfig);
+
+  objectAttrManager_ = simulator_->getObjectAttributesManager();
+  assetAttrManager_ = simulator_->getAssetAttributesManager();
+  sceneAttrManager_ = simulator_->getSceneAttributesManager();
+  physAttrManager_ = simulator_->getPhysicsAttributesManager();
+
+  // configure and initialize default Agent and Sensor
+  auto agentConfig = esp::agent::AgentConfiguration();
+  // TODO: add extra, non-default actions
+  agentConfig.sensorSpecifications[0]->resolution =
+      esp::vec2i(viewportSize[1], viewportSize[0]);
+  // add selects a random initial state and sets up the default controls and
+  // step filter
+  simulator_->addAgent(agentConfig);
+
+  // Set up camera
```

Comment:
I will consider adding more details in the comments. "Why do we need to setup the render camera here since we are using sensor to obtain the observation?"
"Because eventually our visual sensor will render the observations using the renderCamera."

(It let me wonder if it is better to provide an API in the sensor class to set the aspectRatioPolicy for the renderCamera, making the renderCamera completely transparent to the developer. Ideally the user should never, ever know we have a default renderCamera attached in the SG.)

Reply:
Could be part of the VisualSensor API.